plaid/processor.go

```diff
@@ -28,22 +32,29 @@ type createStripeTokenRequest struct {
 	AccountID   string `json:"account_id"`
 }
 
+// CreateStripeTokenResponse defines the unique return format for stripe processor token requests 
 type CreateStripeTokenResponse struct {
 	APIResponse
 	StripeBankAccountToken string `json:"stripe_bank_account_token"`
 }
 
-func (c *Client) createProcessorToken(apiEndpoint, accessToken, accountID string) (resp createProcessorTokenResponse, err error) {
+func (c *Client) requestProcessorToken(apiEndpoint, accessToken, accountID string, processor string) (resp ProcessorTokenResponse, err error) {
 	if accessToken == "" || accountID == "" {
 		return resp, errors.New(apiEndpoint + " - access token and account ID must be specified")
 	}
 
-	jsonBody, err := json.Marshal(createProcessorTokenRequest{
+	requestBody := processorTokenRequest{
 		ClientID:    c.clientID,
 		Secret:      c.secret,
 		AccessToken: accessToken,
 		AccountID:   accountID,
-	})
+	}
+
+	if processor != "" {
```

Comment:
consider adding an explanatory comment here, since the reason for doing this isn't obvious just looking at the `processor` variable itself.

Reply:
alternatively, consider keeping the logic for the per-processor routes separate from the logic for the new generic route. that might make it easier for us down the line if we ever make changes to the generic route but not the others, or vice-versa.